app/code/Magento/Catalog/view/frontend/templates/product/image_with_borders.phtml

```diff
@@ -10,8 +10,9 @@
 /** @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer */
 $width = (int)$block->getWidth();
 $paddingBottom = $block->getRatio() * 100;
+$containerId = uniqid($block->getProductId());
```

Comment:
I believe we should create a function on `Magento\Catalog\Block\Product\Image` to generate the containerId instead of adding it to template.